tests/ga/test_monitor.py

```diff
@@ -546,339 +283,236 @@ def test_reset_loggers_ensuring_timestamp_gets_updated(self, *args):
-                                    is_success=True,
-                                    message="Heartbeat",
-                                    log_event=False)
-
-        # Add agent metric
-        self.event_logger.add_metric("Process", "% Processor Time", "walinuxagent.service", 10)
-
-        # Add agent log
-        self.event_logger.add_log_event(logger.LogLevel.WARNING, "Test sending a log event.")
-
-        # Add extension event file the way extension do it, by dropping a .tld file in the events folder
-        source_file = os.path.join(data_dir, "ext/dsc_event.json")
-        dest_file = os.path.join(conf.get_lib_dir(), "events", "dsc_event.tld")
-        shutil.copyfile(source_file, dest_file)
-
-        # Collect these events and assert they are being sent with the correct sysinfo parameters from the agent
-        with patch.object(protocol, "report_event") as patch_report_event:
-            monitor_handler.collect_and_send_events()
-
-            telemetry_events_list = patch_report_event.call_args_list[0][0][0]
-            self.assertEqual(len(telemetry_events_list.events), 4)
-
-            for event in telemetry_events_list.events:
-                # All sysinfo parameters coming from the agent have to be present in the telemetry event to be emitted
-                for param in monitor_handler.sysinfo:
-                    self.assertTrue(param in event.parameters)
-
-                # The container id, GAVersion are special parameters that are not a part of the static sysinfo parameter
-                # list.
-
-                # The container id value is obtained from the goal state and must be present in all telemetry events.
-                container_id_param = TelemetryEventParam("ContainerId", protocol.client.get_goal_state().container_id)
-                self.assertTrue(container_id_param in event.parameters)
-
-                # Same for GAVersion
-                container_id_param = TelemetryEventParam("GAVersion", CURRENT_AGENT)
-                self.assertTrue(container_id_param in event.parameters)
+        return monitor_handler
+
+    def _create_extension_event(self,
+                               size=0,
+                               name="DummyExtension",
+                               op=WALAEventOperation.Unknown,
+                               is_success=True,
+                               duration=0,
+                               version=CURRENT_VERSION,
+                               message="DummyMessage"):
+        event_data = TestEventMonitoring._get_event_data(name=size if size != 0 else name,
+                op=op,
+                is_success=is_success,
+                duration=duration,
+                version=version,
+                message=random_generator(size) if size != 0 else message)
+        event_file = os.path.join(self.event_dir, "{0}.tld".format(int(time.time() * 1000000)))
+        with open(event_file, 'wb+') as fd:
```

Comment:
Could we use `save_event` here? It handles the naming and `event_dir` creation if needed.

Reply:
save_event should be a private of EventLogger, don't want to introduce dependencies on private members

Reply:
Also, see my comment below